src/renderer/components/modal/swap/pages/done/XmrRedeemInMempoolPage.tsx

```diff
@@ -12,11 +13,11 @@ type XmrRedeemInMempoolPageProps = {
 export default function XmrRedeemInMempoolPage({
   state,
 }: XmrRedeemInMempoolPageProps) {
-  const xmrAmount = useActiveDbState()?.state.Bob.ExecutionSetupDone.state2.xmr;
-  const additionalContent = xmrAmount
-    ? `This transaction transfers ${pionerosToXmr(xmrAmount).toFixed(
-        6
-      )} XMR to ${state?.bobXmrRedeemAddress}`
+  const swap = useActiveSwapInfo();
+  const additionalContent = swap
+    ? `This transaction transfers ${piconerosToXmr(
+        getSwapXmrAmount(swap)
+      ).toFixed(6)} XMR to ${state?.bobXmrRedeemAddress}`
     : null;
```

Comment:
**Suggestion:** Consider adding error handling or a fallback for when `swap` is null or undefined. This will prevent potential errors when trying to access `getSwapXmrAmount(swap)`.
```suggestion
const swap = useActiveSwapInfo() || {};
const additionalContent = getSwapXmrAmount(swap)
  ? `This transaction transfers ${piconerosToXmr(
      getSwapXmrAmount(swap)
    ).toFixed(6)} XMR to ${state?.bobXmrRedeemAddress}`
  : 'Transaction details: Not available';
```